### PR TITLE
feat: sidebar mobile responsiveness tm-59

### PR DESCRIPTION
**Pull request by ShadowOfTheSpace**
- Tablet view
tablet-sidebar

- Mobile view
mobile-sidebar

### CODE REVIEW

File: apps/frontend/src/libs/components/icon/icon.tsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-import { type IconName } from "./libs/types/types.js";
+import { IconName } from "./libs/types/types.js";
+import styles from "./styles.module.css";
 
 type Properties = {
 	name: IconName;
 };
 
-const Icon: React.FC<Properties> = ({ name }: Properties) => {
+const Icon: React.FC<HTMLProps<SVGSVGElement> & Properties> = ({
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/icon/icon.tsx:13]: Farid asked to remove unnecessary props [here](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-trackmates/pull/68#discussion_r1489558417)

File: apps/frontend/src/libs/components/icon/libs/maps/icon-name-to-svg.map.ts [hunk 2]
```diff
@@ -1,11 +1,15 @@
+import BurgerIcon from "~/assets/img/svg/burger.svg?react";
+import HomeIcon from "~/assets/img/svg/home.svg?react";
 import PlusIcon from "~/assets/img/svg/plus.svg?react";
 
-import { type IconName } from "../types/icon-name.type.js";
+import { IconName } from "../types/types.js";
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/icon/libs/maps/icon-name-to-svg.map.ts:5]:
```suggestion
import { type IconName } from "../types/types.js";
```

File: apps/frontend/src/libs/components/icon/libs/types/icon-name.type.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
-type IconName = "plus";
+type IconName = "burger" | "home" | "plus";
 
-export { type IconName };
+export { IconName };
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/icon/libs/types/icon-name.type.ts:3]:
```suggestion
export { type IconName };
```

File: apps/frontend/src/libs/components/app/app.tsx [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import reactLogo from "~/assets/img/react.svg";
-import { Link, Loader, RouterOutlet } from "~/libs/components/components.js";
+import {
+	Link,
+	Loader,
+	RouterOutlet,
+	Sidebar,
+} from "~/libs/components/components.js";
```
Comment (YourTeaChan) [inline apps/frontend/src/libs/components/app/app.tsx:7]: This imports should be from components themselves, not from components.js

**Review by YourTeaChan — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/blurred-background/blurred-background.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	className,
+	isVisible,
+	onClick,
+}: Proprieties) => {
+	if (!isVisible) {
+		return null;
+	}
+
+	return (
+		<button
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/libs/components/blurred-background/blurred-background.tsx:21]: Why do you use html button and not the `Button` component?

**Review by Oleksandra-Masiuk — CHANGES_REQUESTED** [review]

**Review by what1s1ove — CHANGES_REQUESTED** [review]
merge the main branch into your branch because you have a lot of changed that we already have in the main brach

File: apps/frontend/src/libs/components/sidebar/styles.module.css [hunk 6]
```diff
@@ -0,0 +1,142 @@
+.sidebar {
+	z-index: 100;
```
[the remark this answers is not in the document]
Comment (ShadowOfTheSpace) [reply apps/frontend/src/libs/components/sidebar/styles.module.css:2]: I make those indents (99, 100, 101) to be confident that no other elements in the future can be higher then those. Or you advice me to change these values to 1, 2, 3 or something like that?

[the remark this answers is not in the document]
Comment (Oleksandra-Masiuk) [reply apps/frontend/src/libs/components/sidebar/styles.module.css:2]: I am talking about the spaces at the beginning of a code line

[the remark this answers is not in the document]
Comment (Oleksandra-Masiuk) [reply apps/frontend/src/libs/components/sidebar/styles.module.css:2]: yes, probably it is a view bug

File: apps/frontend/src/libs/components/app/app.tsx [hunk 7]
```diff
@@ -1,5 +1,6 @@
-import { Loader, RouterOutlet } from "~/libs/components/components.js";
+import { RouterOutlet } from "~/libs/components/components.js";
 import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import { checkIfSidebarIsShown } from "~/libs/helpers/helpers.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:3]: This helper was removed, check the implementation in the main branch

File: apps/frontend/src/libs/components/button/button.tsx [hunk 8]
```diff
@@ -36,18 +40,18 @@ const Button: React.FC<Properties> = ({
 	);
 
 	const icon = iconName ? <Icon name={iconName} /> : null;
-
+	const labelStyle = hasVisuallyHiddenLabel ? "visually-hidden" : undefined;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/button/button.tsx:43]:
```suggestion
	const labelStyle = getValidClassNames(hasVisuallyHiddenLabel && "visually-hidden");
```

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 9]
```diff
@@ -1,40 +1,71 @@
...
+							key={label}
+							label={label}
+						/>
+					))}
+				</nav>
+			</div>
+			<BlurredBackground
+				className={styles["blurred-background"]}
+				isVisible={isOpen}
+				onClick={handleCloseSidebar}
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:66]:
```suggestion
				onClick={handleToggleSidebar}
```
This component will be present only if `isOpen` is true, that means toggle will make it false

File: apps/frontend/src/libs/components/sidebar/styles.module.css [hunk 10]
```diff
@@ -34,7 +52,94 @@
...
+	.close .menu-item {
+		color: transparent;
+	}
+
+	.title-container {
+		padding-left: 20px;
+	}
+
+	.burger-button {
+		z-index: 101;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/styles.module.css:77]: Move z-index values to global variables. Use 10, 100, 1000

File: apps/frontend/src/libs/enums/app-route.enum.ts [hunk 11]
```diff
@@ -1,5 +1,6 @@
 const AppRoute = {
 	ANY: "*",
+	OVERVIEW: "/overview",
```
Comment (fshabanov) [inline apps/frontend/src/libs/enums/app-route.enum.ts:3]: We don't have /overview, the Overview screen is in ROOT (/)

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by YourTeaChan — APPROVED** [review]

**Review by Ann-Maksay — APPROVED** [review]

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 12]
```diff
@@ -18,11 +18,7 @@ type Properties = {
 const Sidebar: React.FC<Properties> = ({ menuItems }: Properties) => {
 	const [isOpen, setOpen] = useState<boolean>(false);
 
-	const handleCloseSidebar = useCallback(() => {
-		setOpen(false);
-	}, []);
-
-	const toggleOpenSidebar = useCallback(() => {
+	const handleToggleSidebar = useCallback(() => {
 		setOpen(!isOpen);
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:22]:
```suggestion
		setOpen((previous) => !previous);
```

File: apps/frontend/src/libs/components/button/button.tsx [hunk 13]
```diff
@@ -9,9 +9,11 @@ import styles from "./styles.module.css";
 type Properties = {
 	className?: string | undefined;
 	color?: "primary";
+	hasVisuallyHiddenLabel?: boolean;
```
Comment (what1s1ove) [inline apps/frontend/src/libs/components/button/button.tsx:12]: merge main into your branch to avoid unnecessary file changes in pr

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 14]
```diff
@@ -1,40 +1,67 @@
-import logo from "~/assets/img/svg/auth-circle-mobile-logo.svg";
+import React from "react";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:1]:
```suggestion
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by Sytnikovakatya — APPROVED** [review]

File: apps/frontend/src/libs/components/blurred-background/styles.module.css [hunk 15]
```diff
@@ -0,0 +1,15 @@
+.blurred-background {
+	position: fixed;
+	inset: 0;
+	z-index: 99;
+	width: 100%;
+	height: 100%;
+	background-color: #004a40b3;
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/blurred-background/styles.module.css:7]:
```suggestion
	background-color: var(--modal-container-background);
```

File: apps/frontend/src/libs/components/blurred-background/styles.module.css [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+	backdrop-filter: blur(5px);
+	border: none;
+	border-radius: 0;
+}
+
+.blurred-background:hover {
+	background-color: #004a40b3;
```
Comment (GEOFARL) [inline apps/frontend/src/libs/components/blurred-background/styles.module.css:14]:
```suggestion
	background-color: var(--modal-container-background);
```

**Review by MappariChan — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**